mantidproject/mantid
# Notify GDPP on Settings Change

This PR removes the need for the m_settingsChanged flag in the GenericDataProcessorPresenter by making the Reflectometry GUI actively notify it when the settings are changed, this eliminates the need for the `areSettingsChanged` routine and should make it easier to switch the parallelisation model in the `GenericDataProcessorPresenter`.

Additionally it includes the changes from #21364 with the test + compilation failures fixed.
> Exposes the `ReductionType` and `SummationType` options of `ReflectometryReductionOneAuto` on the > Reflectometry GUI settings tab via two comboboxes.

**To test:**
1. Ensure that the Reflectometry Settings tab still correctly passes the settings to the reduction algorithms.
2. Ensure that the SANS v2 GUI continued to function as expected.


Fixes #21353. 

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
